neuralpp/util/util.py

```diff
@@ -613,3 +613,20 @@ def list_for_each(
         if post_index_result is not None:
             post_index_result(index, result)
     return result
+
+
+def check_consistency_of_two_dicts(dict1: Dict, dict2: Dict):
+    intersection_keys = dict1.keys() & dict2.keys()
+    for key in intersection_keys:
+        if dict1[key] != dict2[key]:
+            raise ValueError(f"inconsistent dicts on {key}: {dict1[key]} and {dict2[key]}.")
+
+
+def consistently_update_dict(dict1: Dict, dict2: Dict):
```

Comment:
nit: would `update_consistent_dicts` be a more precise name? It seems "consistent" is modifying "dicts" rather than "update".

Reply:
yes. thx!